web/components/supabase-search.tsx

```diff
@@ -0,0 +1,503 @@
+import { ViewGridIcon, ViewListIcon } from '@heroicons/react/outline'
+import clsx from 'clsx'
+import { Contract } from 'common/contract'
+import { Group } from 'common/group'
+import { debounce, isEqual, uniqBy } from 'lodash'
+import { useRouter } from 'next/router'
+import { createContext, useContext, useEffect, useRef } from 'react'
+import { useEvent } from 'web/hooks/use-event'
+import {
+  historyStore,
+  inMemoryStore,
+  urlParamStore,
+  usePersistentState,
+} from 'web/hooks/use-persistent-state'
+import { useSafeLayoutEffect } from 'web/hooks/use-safe-layout-effect'
+import { track, trackCallback } from 'web/lib/service/analytics'
+import { searchContract } from 'web/lib/supabase/contracts'
+import { safeLocalStorage } from 'web/lib/util/local'
+import { ShowTime } from './contract/contract-details'
+import { ContractsGrid } from './contract/contracts-grid'
+import { ContractsList } from './contract/contracts-list'
+import { groupRoleType } from './groups/group-member-modal'
+import { Col } from './layout/col'
+import { Input } from './widgets/input'
+import { Select } from './widgets/select'
+import { SiteLink } from './widgets/site-link'
+
+const CONTRACTS_PER_PAGE = 20
+
+export const SORTS = [
+  { label: 'Relevance', value: 'relevance' },
+  { label: 'New', value: 'newest' },
+  { label: 'Trending', value: 'score' },
+  { label: 'Daily change', value: 'daily-score' },
+  { label: '24h volume', value: '24-hour-vol' },
+  { label: 'Total traders', value: 'most-popular' },
+  { label: 'High stakes', value: 'liquidity' },
+  { label: 'Last activity', value: 'last-updated' },
+  { label: 'Closing soon', value: 'close-date' },
+  { label: 'Just resolved', value: 'resolve-date' },
+] as const
+
+export type Sort = typeof SORTS[number]['value']
+export const PROB_SORTS = ['prob-descending', 'prob-ascending']
+
+export type filter = 'open' | 'closed' | 'resolved' | 'all'
+
+export type SupabaseSearchParameters = {
+  query: string
+  sort: Sort
+  filter: filter
+  //   facetFilters: SearchOptions['facetFilters']
+}
+
+function getShowTime(sort: Sort) {
+  return sort === 'close-date' || sort === 'resolve-date' ? sort : null
+}
+
+export const INITIAL_STATE = {
```

Comment:
Could add type annotation so it stays in sync: 
`export const INITIAL_STATE: stateType`